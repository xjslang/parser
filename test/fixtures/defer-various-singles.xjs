function testVariousSingleStatements() {
  console.log('testing various single statements')
  
  let counter = 0
  const obj = { value: 10 }
  
  // Sentencia de expresión simple
  defer console.log('simple expression')
  
  // Asignación
  defer counter = 42
  
  // Llamada a método
  defer obj.reset?.()
  
  // Expresión condicional
  defer counter > 0 && console.log('counter is positive:', counter)
  
  // Incremento
  defer counter++
  
  console.log('before defers - counter:', counter)
  console.log('before defers - obj.value:', obj.value)
}

testVariousSingleStatements()
