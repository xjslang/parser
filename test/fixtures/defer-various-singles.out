testing various single statements
before defers - counter: 0
before defers - obj.value: 10
counter is positive: 1
simple expression
